shotgun_api3/shotgun.py

```diff
@@ -1473,20 +1494,33 @@ def update_project_last_accessed(self, project, user=None):
         record = self._call_rpc("update_project_last_accessed_by_current_user", params)
         result = self._parse_records(record)[0]
 
-
-    def _get_session_token(self):
-        """Hack to authenticate in order to download protected content
-        like Attachments
+    def generate_session_token(self):
```

Comment:
hehe, i was actually using `generate` to try to emphasize that it is doing something on the server - not just getting some existing data, but in fact fabricating a new record. `get_session_token` sounded more to me like you are getting an existing token - the idea with generate was to imply that a new one is being returned. (confusingly, also noticed that the there is a `get_attachment_download_url()` method as the only other "getter" in the shotgun API and that isn't actually talking to the server!).

Do you think it's important to match the server side call? I thought this was more of a private detail?

I don't have a strong opinion at all, get is totally fine too, let me know if you think I should change it!